include/cudaq/Optimizer/Dialect/Quake/QuakeOps.h

```diff
@@ -70,6 +70,15 @@ void genericOpPrinter(mlir::OpAsmPrinter &_odsPrinter, mlir::Operation *op,
 // Utility functions to test the form of an operation.
 //===----------------------------------------------------------------------===//
 
+// Is \p op in the Quake dialect?
+// TODO: Is this StringRef comparison faster than calling MLIRContext::
+// getLoadedDialect("quake")?
```

Comment:
I believe it must be. This does a O(c) —where c is at most 5—comparison on two character sequences. The getLoadedDialect call does an O(log_2(n)) search over all loaded dialects with the same constant factors.

Reply:
Gotcha, I brought the comment over with the rest of the function so I didn't know.